src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubBuildStatusNotification.java

```diff
@@ -239,35 +236,12 @@ private static GitHubSCMSource getSCMSource(final SCMSourceOwner scmSourceOwner)
 
     private static String resolveHeadCommit(GHRepository repo, SCMRevision revision) throws IllegalArgumentException {
         if (revision instanceof SCMRevisionImpl) {
-            SCMRevisionImpl rev = (SCMRevisionImpl) revision;
-            try {
-                GHCommit commit = repo.getCommit(rev.getHash());
-                List<GHCommit> parents = commit.getParents();
-                // if revision has two parent commits, we have really a MergeCommit
-                if (parents.size() == 2) {
-                    SCMHead head = revision.getHead();
-                    // MergeCommit is coming from a pull request
-                    if (head instanceof PullRequestSCMHead) {
-                        GHPullRequest pullRequest = repo.getPullRequest(((PullRequestSCMHead) head).getNumber());
-                        for (GHPullRequestCommitDetail commitDetail : pullRequest.listCommits()) {
-                            if (commitDetail.getSha().equals(parents.get(0).getSHA1())) {
-                                // Parent commit (HeadCommit) found in PR commit list
-                                return parents.get(0).getSHA1();
-                            }
-                        }
-                        // First parent commit not found in PR commit list, so returning the second one.
-                        return parents.get(1).getSHA1();
-                    } else {
-                        return rev.getHash();
-                    }
-                }
-                return rev.getHash();
-            } catch (IOException e) {
-                LOGGER.log(Level.WARNING, null, e);
-                throw new IllegalArgumentException(e);
-            }
+            return ((SCMRevisionImpl) revision).getHash();
+        } else if (revision instanceof PullRequestSCMRevision) {
+            return ((PullRequestSCMRevision) revision).getPullHash();
+        } else {
+            throw new IllegalArgumentException("did not recognize " + revision);
```

Comment:
By keeping a reasonable representation of the `SCMRevision`s to begin with, we avoid the need to try to reconstruct what we just built.